engine/util/util.go

```diff
@@ -331,3 +331,40 @@ func unitStateAction(ctx context.Context, conn *dbus.Conn, unit, action string)
 	}
 	return nil
 }
+
+// edgeCombiner holds the state of the auto edge generator for combining edges.
+type edgeCombiner struct {
+	ae  []engine.AutoEdge
+	ptr int
+}
+
+// Next returns the next automatic edge.
+func (obj *edgeCombiner) Next() []engine.ResUID {
+	if len(obj.ae) <= obj.ptr {
+		panic("shouldn't be called anymore!")
+	}
+	return obj.ae[obj.ptr].Next() // return the next edge
+}
+
+// Test takes the output of the last call to Next() and outputs true if we
+// should continue.
+func (obj *edgeCombiner) Test(input []bool) bool {
+	if len(input) == 0 {
```

Comment:
Is this needed? If anything, you'd *maybe* want something as well in Next() if you had this in case someone ran an empty struct.

Reply:
I guess not. If you run an empty struct you just don't get any auto edges. If you run a nil struct it panics anyway, so I don't have to check that either.